### ISSUE TITLE
How the available resource of member cluster is accessible ?

### ISSUE THREAD

**Issue by Fatimastm** [karmada-io/karmada#3131]
**Please provide an in-depth description of the question you have**:
How the available (not allocatable) amount of cpus and memory of each members are accessible? I guess they are accessible in clusterv1alpha1 but not sure if I can trace or force karmada to print to logger.

**Comment by XiShanYongYe-Chang** [comment]
Are you referring to this:
https://github.com/karmada-io/karmada/blob/59227f0d4a3b9cd3d79485984202e99c3aa302df/pkg/apis/cluster/v1alpha1/types.go#L331-L351

> but not sure if I can trace or force karmada to print to logger.

Sorry, I misunderstand this question. What is your use scenario?

**Comment by RainbowMango** [comment]
> I guess they are accessible in clusterv1alpha1 but not sure if I can trace or force karmada to print to logger.

Yeah, you can get the available resource from `Cluster` objects by the REST API. 
For now, Karmada doesn't print these logs. I wonder why you want to get that info from logs?

**Comment by Fatimastm** [comment]
> Yeah, you can get the available resource from `Cluster` objects by the REST API.
> For now, Karmada doesn't print these logs. I wonder why you want to get that info from logs?

Thanks. Actually I am curios to see this value afer each deployment. For instance, I want to define a specific number of core in the deployment of pod, then I would like to see which clusters have enough core.

**Comment by Fatimastm** [comment]
> Are you referring to this:
> 
> https://github.com/karmada-io/karmada/blob/59227f0d4a3b9cd3d79485984202e99c3aa302df/pkg/apis/cluster/v1alpha1/types.go#L331-L351
> 

Yes. I referred to that. 

> > but not sure if I can trace or force karmada to print to logger.
> 
> Sorry, I misunderstand this question. What is your use scenario?

For instance, I want to define a specific number of core in the deployment of pod, then I would like to see which clusters have enough core.

**Comment by RainbowMango** [comment]
> For instance, I want to define a specific number of core in the deployment of pod, then I would like to see which clusters have enough core.

I see. But, just a reminder the `Allocatable ` is the summation of resources on `all nodes`, it's not very accurate by using it to determine whether a Pod could be set, the `AllocatableModelings` will be more precise.

**Comment by Fatimastm** [comment]
> I see. But, just a reminder the `Allocatable ` is the summation of resources on `all nodes`, it's not very accurate by using it to determine whether a Pod could be set, the `AllocatableModelings` will be more precise.

That's right. Thanks. so for tracing, is nodesummery accessible with something like clusterv1alpha.cpu?

**Comment by jwcesign** [comment]
> That's right. Thanks. so for tracing, is nodesummery accessible with something like clusterv1alpha.cpu?
Yes, you can do in this way. For better scheduling, `AllocatableModelings` is better way.

**Comment by RainbowMango** [comment]
Yes, both `nodeSummary` and `resourceSummary` are accessible by leveraging the REST API.

**Comment by Fatimastm** [comment]
> Yes, both `nodeSummary` and `resourceSummary` are accessible by leveraging the REST API.

Thanks @RainbowMango. Sorry for my delay. I was working on implementation.
After creating rest api, I tested in a function outside karmada and tried to retrive list of node in memebr1:
`nodes, err := my_clientApi.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})` .
and got :

> cluster member1 has node: &NodeList{ListMeta:{   <nil>},Items:[]Node{},}

I am wondering if there is  necessary RBAC permissions to access the nodes of member clusters. 
I mean, is there any changes for access to a k8s cluster api after if this cluster joined the karmada? 
I did some tests with normal cluster and the list was not empty.

**Comment by jwcesign** [comment]
I think you need to use this kind of API(referring [link](https://karmada.io/docs/userguide/bestpractices/unified-auth/#step4-grant-permission-to-serviceaccount-in-member1-cluster)):
```sh
kubectl --kubeconfig tom.config get --raw /apis/cluster.karmada.io/v1alpha1/clusters/member1/proxy/api/v1/nodes
``

**Comment by RainbowMango** [comment]
Hi @Fatimastm 
Given what you are asking in this issue is about `How to retrieve available resource of member cluster`, so you need to try to get the `Clusters` resource instead of `Nodes`, you should use [this client](https://github.com/karmada-io/karmada/blob/0652cfbe2c87946e2c6ca8ab3a6d5f22ff9a3740/pkg/generated/clientset/versioned/clientset.go#L23).

By the way, the Karmada control plane doesn't really manage `Nodes`, that's the reason you got nothing.
